### DISCUSSION TITLE
Possible Regression with Safari Password Autofill and Alpine x-model (Safari 16.3 / Alpine >3.10.5)

### DISCUSSION THREAD

**Discussion by kylecotter** [alpinejs/alpine#3454] [7. Bugs (archived)]
Hey all,

Not sure if what was addressed in https://github.com/alpinejs/alpine/pull/1969 has since regressed.

Using a simple test of a disabled submit button until username/password fields have value, what was working in Alpine 3.10.5 has stopped working as of Alpine 3.11.0 and continues to not work in Alpine 3.12.0.

Safari 16.3 Password Autofill and Alpine x-model - Working (Alpine 3.10.5) - https://codepen.io/kylecotter/pen/rNZYBrW
Safari 16.3 Password Autofill and Alpine x-model - Not Working (Alpine 3.12.0) - https://codepen.io/kylecotter/pen/jOvaNzN

Steps to reproduce:
1. Open Codepen in Safari (16.3 was what's been tested)
2. Note how the older version of Alpine works as expected by the button turning blue upon autofill
3. The newer version of Alpine does not remove the disabled state after autofill

Thank you for any insight!

**Comment by madsem** [comment]
I'm looking for a solution to the same issue, running into this as well. 

But autocomplete in general isn't working anymore, email fields as well...

I upgraded Alpine to 3.12, on `Safari Version 15.5` before it was working without issues.

PS:

Downgraded to 3.10.2 and this field, works again with safari autocomplete:

```html
<div class="not-prose">
    <input type="email" 
    name="email" 
    id="email" 
    data-validation='["required", "email"]'
    @keyup.enter="!completed && await forward"
    autocomplete="email"
    class="valid block w-full shadow-sm rounded-md border-gray-300 p-6 text-sm lg:text-base placeholder:text-slate-500 placeholder:text-base" 
    placeholder="johndoe@gmail.com"
    x-model="leadFields.email"
    required
    autofocus>
    <small class="hidden error-text"></small>
</div>
```

**Comment by ekwoka** [comment]
The most likely cause I think would be in this commit

https://github.com/alpinejs/alpine/commit/7628c62283116157714f4e6e39236868712547fa

The rest in that time period are very minor and shouldn't cause the issue found. I don't see what here might be responsible, but it's more a guess just based on the extra levels of abstraction here, seeing what might be the cause.

**Comment by kylecotter** [comment]
Solved via #3483. More context: #3482
